src/components/TokenEnterAmount.tsx

```diff
@@ -35,6 +41,200 @@ function groupNumber(value: string) {
   return value.replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, 'group')
 }
 
+function roundTokenAmount(value: string, token: TokenBalance) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  if (value === '') {
+    return ''
+  }
+
+  const bigNum = parseInputAmount(value, decimalSeparator)
+
+  if (bigNum.isLessThan(0.000001)) {
+    return `<0${decimalSeparator}000001 ${token.symbol}`
+  }
+
+  const grouped = groupNumber(bigNum.decimalPlaces(6).toString())
+    .replaceAll('.', decimalSeparator)
+    .replaceAll('group', groupingSeparator)
+  return `${grouped} ${token.symbol}`
+}
+
+function roundLocalAmount(value: string, localCurrencySymbol: LocalCurrencySymbol) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  if (value === '') {
+    return ''
+  }
+
+  const bigNum = parseInputAmount(value, decimalSeparator)
+
+  if (bigNum.isLessThan(0.000001)) {
+    return `<${localCurrencySymbol}0${decimalSeparator}000001`
+  }
+
+  const rounded = bigNum.isLessThan(0.01) ? bigNum.toPrecision(1) : bigNum.toFixed(2)
+  const grouped = groupNumber(rounded.toString())
+    .replaceAll('.', decimalSeparator)
+    .replaceAll('group', groupingSeparator)
+  return `${localCurrencySymbol}${grouped}`
+}
+
+export function useEnterAmount(props: {
+  token: TokenBalance
+  onSelectToken?: (token: TokenBalance) => void
+}) {
+  const { decimalSeparator, groupingSeparator } = getNumberFormatSettings()
+  const inputRef = useRef<RNTextInput>(null)
+  const bottomSheetRef = useRef<BottomSheetModalRefType>(null)
+  const [amount, setAmount] = useState('')
+  const [amountType, setAmountType] = useState<AmountEnteredIn>('token')
+
+  // this should never be null, just adding a default to make TS happy
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol) ?? LocalCurrencySymbol.USD
+  const tokenInfo = useTokenInfo(props.token.tokenId)
+  const usdToLocalRate = useSelector(usdToLocalCurrencyRateSelector)
+
+  const amountRaw = useMemo(() => {
+    return amount.replaceAll(groupingSeparator, '').replaceAll(decimalSeparator, '.')
+  }, [amount])
+
+  const derived = useMemo(() => {
```

Comment:
I didn't really know how to call this but I didn't want to destruct `token/local` as they are quite too "general" names. Calling it `derived` points out that this is a derived data so I thought it makes sense to keep it that way. 

I'm open for suggestions though!

Reply:
ah, naming is hard 🙈 what do you think about `parsedAmounts` or `formattedAmounts`?

Reply:
@kathaypacific that's the problem - it includes both parsed and formatted amount 😄
(unless we perceive formatting as a part of "parsing"?)

Do you think any of these will work?
- `calculations`
- `calculatedAmounts`
- `processedAmounts` (I am leaning towards this one)